services/QuillLMS/app/models/stripe_integration/subscription.rb

```diff
@@ -14,32 +16,38 @@ def stripe_subscription_id
       return nil if stripe_invoice_id.nil?
 
       stripe_invoice.subscription
-    rescue Stripe::InvalidRequestError
-      nil
     end
 
-    def stripe_customer_id
-      return nil if stripe_invoice_id.nil?
+    private def stripe_card
+      stripe_payment_method || stripe_source
+    end
 
-      stripe_subscription.customer
+    private def stripe_customer
+      Stripe::Customer.retrieve(stripe_customer_id)
     rescue Stripe::InvalidRequestError
       nil
     end
 
-    private def stripe_customer
-      Stripe::Customer.retrieve(stripe_customer_id)
+    private def stripe_customer_id
+      stripe_subscription.customer
     end
 
     private def stripe_invoice
       Stripe::Invoice.retrieve(stripe_invoice_id)
     end
 
-    private def stripe_default_payment_method_id
-      stripe_subscription.default_payment_method
+    private def stripe_payment_method
+      Stripe::PaymentMethod.retrieve(stripe_payment_method_id)&.card
+    rescue Stripe::InvalidRequestError
+      nil
+    end
+
+    private def stripe_payment_method_id
+      stripe_subscription.default_payment_method || stripe_customer.invoice_settings.default_payment_method
     end
 
-    private def stripe_payment_method
-      Stripe::PaymentMethod.retrieve(stripe_default_payment_method_id)
+    private def stripe_source
+      Stripe::PaymentMethod.list(customer: stripe_customer_id, type: 'card')&.data&.first
```

Comment:
It might not matter if there should only be one card in the list, but if there could be multiple, should this be `.last` to get a newer card(I'm not 100% sure how they would be ordered, I'm guessing newest would be last, but that could be wrong)

Reply:
This is a good point.  I was basically reusing [old](https://github.com/empirical-org/Empirical-Core/blob/f215d13bcb43cb26f56dc3bf853154acdba55ed6/services/QuillLMS/app/models/user.rb#L300) code for source, but what we really need here is `default_source` since that would resolve the case where there are two sources and the older card for whatever reason is the default one.